src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -90,6 +110,7 @@ private void clearCommand() {
 
 	private void runCommand() {
 		commandManager.runCurrentCommand();
+		CommandHistoryController.addNewHistoryEntry(CommandsFeature.getDriverCommandManager().getCommandList(),CommandsFeature.getDriverCommandManager().getCommandName());
```

Comment:
commandManager.runCurrentCommand() can be executed from other places. addNewHistoryEntry should be triggered by this command.